src/main/java/com/capstone/doctorsOffice/controllers/DoctorController.java

```diff
@@ -0,0 +1,45 @@
+package com.capstone.doctorsOffice.controllers;
+
+import com.capstone.doctorsOffice.dtos.DoctorDto;
+import com.capstone.doctorsOffice.repositories.DoctorRepository;
+import com.capstone.doctorsOffice.services.DoctorService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/api/doctor")
+public class DoctorController {
+    @Autowired
+    private DoctorService doctorService;
+    @Autowired
+    private DoctorRepository doctorRepository;
+
+    @GetMapping("")
+    public List<DoctorDto> getAllDoctors() {
+        return doctorService.getAllDoctors();
+    }
+
+    @GetMapping("/{id}")
+    public Optional<DoctorDto> getDoctorById(@PathVariable Long id) {
+        return doctorService.getDoctorById(id);
+    }
+
+    @PostMapping("/new-doctor")
+    public List<String> addNewDoctor(@RequestBody DoctorDto doctorDto) {
+        return doctorService.addNewDoctor(doctorDto);
+    }
+
+    @DeleteMapping("/{id}")
+    public void deleteDoctor(@PathVariable Long id) {
+        doctorService.deleteDoctor(id);
+    }
+
+    @PutMapping("/{id}")
+    public void updateDoctor(@PathVariable Long id, @RequestBody DoctorDto doctorDto) {
+        doctorService.updateDoctor(doctorDto);
+    }
```

Comment:
You should pass the PathVariable `id` the the `updateDoctor` method in the service and use that id to look up the doctor.

```
doctorService.updateDoctor(id, doctorDto);
```

You would need to then update the function signature in `DoctorService` and `DoctorServiceImpl`